sdks/go/pkg/beam/io/fhirio/common.go

```diff
@@ -91,18 +108,53 @@ func (c *fhirStoreClientImpl) search(storePath, resourceType string, queries map
 	}
 
 	searchRequest := &healthcare.SearchResourcesRequest{}
+	fhirService := c.healthcareService.Projects.Locations.Datasets.FhirStores.Fhir
+
 	if resourceType == "" {
-		return c.fhirService.Search(storePath, searchRequest).Do(queryParams...)
+		return fhirService.Search(storePath, searchRequest).Do(queryParams...)
 	}
-	return c.fhirService.SearchType(storePath, resourceType, searchRequest).Do(queryParams...)
+	return fhirService.SearchType(storePath, resourceType, searchRequest).Do(queryParams...)
 }
 
-func newFhirStoreClient() *fhirStoreClientImpl {
-	healthcareService, err := healthcare.NewService(context.Background(), option.WithUserAgent(UserAgent))
+func (c *fhirStoreClientImpl) deidentify(srcStorePath, dstStorePath string, deidConfig *healthcare.DeidentifyConfig) (operationResults, error) {
+	deidRequest := &healthcare.DeidentifyFhirStoreRequest{
+		Config:           deidConfig,
+		DestinationStore: dstStorePath,
+	}
+	operation, err := c.healthcareService.Projects.Locations.Datasets.FhirStores.Deidentify(srcStorePath, deidRequest).Do()
 	if err != nil {
-		panic("Failed to initialize Google Cloud Healthcare Service. Reason: " + err.Error())
+		return operationResults{}, err
+	}
+	return c.pollTilCompleteAndCollectResults(operation)
+}
+
+func (c *fhirStoreClientImpl) pollTilCompleteAndCollectResults(operation *healthcare.Operation) (operationResults, error) {
+	var err error
+	for !operation.Done {
+		time.Sleep(15 * time.Second)
+		operation, err = c.healthcareService.Projects.Locations.Datasets.Operations.Get(operation.Name).Do()
```

Comment:
Yes. When the healthcare client executes `Deidentify`, it returns a `healthcare.Operation` which represents a long-running operation that is the result of a network API call. While the operation's `Done` member variable is not set to true, it means the operation is still in progress and we need to poll until it is complete.

Reply:
Is it possible for the operation to be Done immediately then? If not, it seems like a good idea to poll once right before the loop.

Reply:
Yea, after inspecting the source code of the Deidentify client function, it seems like it could be Done immediately. Updated to poll once before the loop.